types/node/package.json

```diff
@@ -2,15 +2,20 @@
     "private": true,
     "types": "index",
     "typesVersions": {
-        ">=3.5.0-0": {
+        ">=3.7.0-0": {
             "*": [
-                "ts3.5/*"
+                "ts3.7/*"
             ]
         },
         ">=3.2.0-0": {
             "*": [
                 "ts3.2/*"
             ]
+        },
+        ">=3.5.0-0": {
```

Comment:
The 3.5 check should come before the 3.2 check, though that primarily relies on for-in enumeration ordering of the keys of the object. A more reliable mechanism would be to use non-overlapping ranges:

```json
{
  "typesVersions": {
    ">=3.7.0.0-0": { ... },
    ">=3.5.0.0-0 <3.7.0.0-0": { ... },
    ">=3.2.0.0-0 <3.5.0.0-0": { ... }
  }
}
```

Reply:
Travis doesn't seem to like that:

```
2> node START
2> node failing:
2> AssertionError [ERR_ASSERTION]: "typesVersions" in '/home/travis/build/DefinitelyTyped/DefinitelyTyped/types/node/package.json' is not set right. Should be: {
2>     ">=3.2.0-0": {
2>         "*": [
2>             "ts3.2/*"
2>         ]
2>     },
2>     ">=3.5.0-0": {
2>         "*": [
2>             "ts3.5/*"
2>         ]
2>     },
2>     ">=3.7.0-0": {
2>         "*": [
2>             "ts3.7/*"
2>         ]
2>     }
2> }
2>     at Object.<anonymous> (/home/travis/build/DefinitelyTyped/DefinitelyTyped/node_modules/dtslint/bin/checks.js:35:20)
2>     at Generator.next (<anonymous>)
2>     at fulfilled (/home/travis/build/DefinitelyTyped/DefinitelyTyped/node_modules/dtslint/bin/checks.js:5:58)
```

Reply:
I've reverted this change for now so that the build passes, but adjusted the order in `package.json`.

I've made a [revert commit](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/42786/commits/c6c2a4ce6586a86ada902276e3ec9d05c267d3be) so that you can review the changes, which I'll drop before this branch is merged.